src/avram/validations.cr

```diff
@@ -1,39 +1,96 @@
-require "./validations/**"
+require "./validations/callable_error_message"
 
+# A number of methods for validating Avram::Attributes
+#
+# This module is included in `Avram::Operation` and `Avram::SaveOperation`
 module Avram::Validations
-  private def validate_required(*attributes, message = "is required")
+  extend self
+
+  # Validates that the passed in attributes have values
+  #
+  # You can pass in one or more attributes at a time. The attribute will be
+  # marked as invalid if the value is `nil`, or "blank" (empty strings or strings with just whitespace)
+  #
+  # `false` is not considered invalid.
+  #
+  # ```
+  # validate_required name, age, email
+  # ```
+  def validate_required(*attributes, message : Avram::Attribute::ErrorMessage = "is required")
     attributes.each do |attribute|
       if attribute.value.blank? && attribute.value != false
         attribute.add_error message
       end
     end
   end
 
-  private def validate_acceptance_of(attribute : Attribute(Bool?), message = "must be accepted")
+  # Validate whether an attribute was accepted (`true`)
+  #
+  # This validation is only for Boolean Attributes. The attribute qill be marked
+  # as invalid for any value other than `true`.
+  def validate_acceptance_of(attribute : Avram::Attribute(Bool?), message : Avram::Attribute::ErrorMessage = "must be accepted")
     if attribute.value != true
       attribute.add_error message
     end
   end
 
-  private def validate_confirmation_of(attribute, with confirmation_attribute, message = "must match")
+  # Validates that the values of two attributes are the same
+  #
+  # Takes two attributes and if the values are different the second attribute
+  # (`with`/`confirmation_attribute`) will be marked as invalid
+  #
+  # Example:
+  #
+  # ```
+  # validate_confirmation_of password, with: password_confirmation
+  # ```
+  #
+  # If `password_confirmation` does not match, it will be marked invalid.
+  def validate_confirmation_of(
+    attribute : Avram::Attribute(T),
+    with confirmation_attribute : Avram::Attribute(T),
```

Comment:
Make sure the Attribute types are the same. So you can't accidentally compare  a Bool to a String